File: frontend/src/types/research.ts
// Mirrors backend/app/schemas/research.py (architecture §9, §26 Phase 4).

export type MacroObservationOut = {
  series_key: string;
  value: string; // Decimal -> JSON string; see market_valuation.ts's note
  unit: string;
  region: string;
  provider: string;
  observed_at: string;
};

export type ResearchItemOut = {
  title: string;
  summary: string;
  source_name: string;
  source_url: string;
  published_at: string | null;
  retrieved_at: string;
};

export type MacroSnapshotOut = {
  available: boolean;
  as_of: string | null;
  observations: MacroObservationOut[];
  narrative_items: ResearchItemOut[];
  reason: string | null;
};

export type SectorResearchOut = {
  available: boolean;
  sector: string;
  as_of: string | null;
  items: ResearchItemOut[];
  reason: string | null;
};

export type ResearchRunOut = {
  id: string;
  type: string;
  sector: string | null;
  status: string;
  started_at: string;
  completed_at: string | null;
  methodology_version: string;
  item_count: number;
  error_message: string | null;
};

File: backend/app/schemas/research.py
"""Pydantic schemas for the research API (app/api/research.py)."""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ResearchItemOut(BaseModel):
    title: str
    summary: str
    source_name: str
    source_url: str
    source_type: str
    published_at: datetime | None
    retrieved_at: datetime


class MacroResearchOut(BaseModel):
    available: bool
    as_of: datetime | None
    items: list[ResearchItemOut]
    reason: str | None = None


class SectorResearchOut(BaseModel):
    available: bool
    sector: str
    as_of: datetime | None
    items: list[ResearchItemOut]
    reason: str | None = None


class CompanyResearchOut(BaseModel):
    available: bool
    holding_id: UUID
    ticker: str
    as_of: datetime | None
    items: list[ResearchItemOut]
    reason: str | None = None


class ResearchRunOut(BaseModel):
    id: UUID
    type: str
    sector: str | None
    holding_id: UUID | None
    status: str
    started_at: datetime
    completed_at: datetime | None
    methodology_version: str
    item_count: int
    error_message: str | None
